front-end/src/components/CustomInput.tsx

```diff
@@ -0,0 +1,162 @@
+    feedbackVariation,
+    first,
+    feedback,
+    align,
+    placeholder,
+    prompt,
+    modeID,
+    questionMode,
+    setText,
+    text,
+  } = useInputContext[inputID]
+
+  // To select the first created input field on the site
+  const firstInputRef = useRef<HTMLInputElement | null>(null)
+  useEffect(() => {
+    if (firstInputRef.current) {
+      firstInputRef.current.focus()
+    }
+  }, [inputID])
+
+  const inputBorderColor = modeID === "invalid" ? "border-red-500 focus:border-red-500" : ""
+
+  const alignDiv: string = "w-full h-full"
+
+  let promptElement
+  if (prompt) {
+    promptElement = <Markdown md={prompt} />
+  }
+
+  let spacing
+  if (align === "NL") {
+    spacing = <br />
+  }
+
+  const [isInputFocused, setIsInputFocused] = useState(false)
+  let inputElement
+  // constantly below the input field
+  if (feedbackVariation === "below") {
+    inputElement = (
+      <div>
+        {spacing}
+        <div className="flex flex-col">
+          <div className="flex flex-row items-center">
+            <div className="mr-2">{promptElement}</div>
+            <div className={`${alignDiv} relative`}>
+              <Input
+                ref={first ? firstInputRef : null}
+                key={`newline-input-${inputID}`}
+                autoFocus
+                disabled={questionMode === "correct" || questionMode === "incorrect"}
+                value={text || ""}
+                onChange={(e) => {
+                  setText ? setText(e.target.value) : ""
+                }}
+                type="text"
+                className={`${inputBorderColor} focus:outline-none`}
+                placeholder={placeholder || ""}
+              />
+            </div>
+          </div>
+          <FeedbackComponent
+            formatFeedback={feedback ? feedback : ""}
+            mode={modeID ? modeID : "draft"}
+            type={"below"}
+          />
+        </div>
+        {spacing}
+      </div>
+    )
+  } else {
+    inputElement = (
+      <div>
+        {spacing}
+        <div className="flex flex-row items-center">
+          <div className="mr-2">{promptElement}</div>
+          <div className={`${alignDiv} relative`}>
+            <Input
+              ref={first ? firstInputRef : null}
+              key={`newline-input-${inputID}`}
+              autoFocus
+              disabled={questionMode === "correct" || questionMode === "incorrect"}
+              value={text || ""}
+              onChange={(e) => {
+                setText ? setText(e.target.value) : ""
+              }}
+              onFocus={() => setIsInputFocused(true)}
+              onBlur={() => setIsInputFocused(false)}
+              type="text"
+              className={`${inputBorderColor} mb-1 w-full focus:outline-none`}
+              placeholder={placeholder || ""}
+            />
+            {isInputFocused && text && feedback && (
+              <FeedbackComponent
+                formatFeedback={feedback ? feedback : ""}
```

Comment:
kann zu `feedback ?? ""` verkürzt werden.

Reply:
I check if `text` is not empty, because otherwise the user still has to type something. It's not necessary, because the generator should return an empty string if the text is empty. I'll take it out. 

But I would keep `isInputFocused`